LanguageFeatures/Class-modifiers/grammar_A01_t01.dart

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion The grammar is:
+///
+/// classDeclaration  ::= (classModifiers | mixinClassModifiers) 'class' typeIdentifier
+///                       typeParameters? superclass? interfaces?
+///                       '{' (metadata classMemberDeclaration)* '}'
+///                       | classModifiers 'class' mixinApplicationClass
+///
+/// classModifiers    ::= 'sealed'
+///                     | 'abstract'? ('base' | 'interface' | 'final')?
+///
+/// mixinClassModifiers ::= 'abstract'? 'base'? 'mixin'
+///
+/// mixinDeclaration  ::= 'base'? 'mixin' typeIdentifier typeParameters?
+///                       ('on' typeNotVoidList)? interfaces?
+///                       '{' (metadata classMemberDeclaration)* '}'
+///
+/// @description Check that it is a compile-time error if any of
+/// keywords/built-in identifiers `abstract`, `final`, `interface`, `mixin`, or
+/// `class` is used as an identifier
```

Comment:
'keyword' is not specified, that's just a loose word which can be used when we want to give a hint that we are talking about a special word (and it could include `hide` and other words which are actually just plain identifiers, but occur somewhere in the grammar).

`abstract`, `interface`, `mixin`, are built-in identifiers.
`final`and `class` are reserved words.

So `final` and `class` cannot be an identifier at all, but `abstract`/`interface`/`mixin` can be the names of everything except types.

There is no need to test the treatment of reserved words, they have been unchanged for many years, and they have been tested before this feature. Similarly, there is no need to test `abstract` or `mixin`, that's been done earlier.

So we should test `interface` as a built-in identifier, it's a new thing that we are actually using that word in the grammar.

That's a small test, but that's OK. ;-)